### PR TITLE
Propagate changes to backend loading to setup/setupext.

### PR DESCRIPTION
**Pull request by anntzer**
1) Restore the ability to set the default backend from setup.cfg (finishes closing #11844 attn @tacaswell).
2) Remove the backend fallback code.

Note that not including `mpl-data/*.glade` in the package data is fine:
that file had been removed in 962388f anyways.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
Thanks @anntzer !

File: setup.cfg.template [hunk 1]
```diff
@@ -81,10 +75,9 @@
 #
 # The Agg, Ps, Pdf and SVG backends do not require external dependencies. Do
 # not choose MacOSX, or TkAgg if you have disabled the relevant extension
-# modules.  Agg will be used by default.
+# modules.  The default is determined by fallback.
```
Comment (mdboom) [inline setup.cfg.template]: I don't understand this change.  Isn't the fallback approach being removed by this PR?  (Or is this referring to something else recent that I probably missed while away from the project for the last couple of years...)

**Comment by mdboom** [comment]
For those building from source (which I'm sure is a decreasing number), this is bound to cause confusion because it will no longer attempt to pick the "best" backend for their system.

For packagers, just telling them to set things with `setup.cfg` (which has worked for a long time, but wasn't the default) is clearly a step forward.

I think optimizing for the common case, this makes sense.  But maybe to make the first faction happy we should say something at the end of compilation like: "Your default backend is set to X.  This may not be optimal for your system."?

**Comment by anntzer** [comment]
Indeed, things changed with https://github.com/matplotlib/matplotlib/pull/9795 / https://github.com/matplotlib/matplotlib/pull/11581 / https://github.com/matplotlib/matplotlib/pull/11600 (sorry the discussion is a bit spread out).  We still pick the best backend, but do it at import time now.  So the best-backend-picking behavior occurs even when you install from a prebuilt distribution (wheel, linux distro, etc.).

**Comment by tacaswell** [comment]
This has infact passed on travis, a webhook just got lost.

File: setup.cfg.template (hunk 1 above)
Comment (tacaswell) [reply setup.cfg.template]: The fallback has been moved to runtime (not build time).

**Pull request merged**